src/storage/conversion/SimpleTurtleQuadConverter.ts

```diff
@@ -0,0 +1,91 @@
+import { matchingMediaType } from '../../util/Util';
+import { PassThrough } from 'stream';
+import { Representation } from '../../ldp/representation/Representation';
+import { RepresentationConverter } from './RepresentationConverter';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import { RepresentationPreferences } from '../../ldp/representation/RepresentationPreferences';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import { CONTENT_TYPE_QUADS, DATA_TYPE_BINARY, DATA_TYPE_QUAD } from '../../util/ContentTypes';
+import { StreamParser, StreamWriter } from 'n3';
+
+/**
+ * Converts `text/turtl`e to/from `internal/quads`.
+ * Simple because this could easily support many other types (everything supported by N3.js).
+ * Will error if the Representation has no contentType or none of the preferences matches one of the two types.
+ */
+export class SimpleTurtleQuadConverter extends RepresentationConverter {
+  public async canHandle(input: { representation: Representation; preferences: RepresentationPreferences }):
+  Promise<void> {
+    const inType = input.representation.metadata.contentType;
+    if (!inType) {
+      throw new UnsupportedHttpError('Input type required for conversion.');
+    }
+    this.bestMatchingType(input.preferences);
+    if (!matchingMediaType(inType, 'text/turtle') && !matchingMediaType(inType, CONTENT_TYPE_QUADS)) {
+      throw new UnsupportedHttpError('Can only convert from turtle and quads.');
```

Comment:
I would actually split this in two classes then. One for Turtle, one for quads. Is there a reason they are merged?

Reply:
No specific reason besides that they both use quads on one end of the transformation and turtle on the other. Can split up to a turtle -> quads converter and a quads -> turtle converter.

Reply:
i would split up then indeed.